test/server/content-charset-test.js

```diff
@@ -0,0 +1,287 @@
+                var contentType;
+                var src;
+
+                switch (req.params.resourceType) {
+                    case 'stylesheet':
+                        contentType = 'text/css';
+                        src         = stylesheetSrc;
+                        break;
+                    case 'manifest':
+                        contentType = 'text/cache-manifest';
+                        src         = manifestSrc;
+                        break;
+                }
+
+                switch (req.params.charsetType) {
+                    case 'bom':
+                        res.set('content-type', contentType + '; charset=' + charsetContentType);
+                        res.end(iconv.encode(src, charsetBOM, { addBOM: true }));
+                        break;
+                    case 'content-type':
+                        res.set('content-type', contentType + '; charset=' + charsetContentType);
+                        res.end(iconv.encode(src, charsetContentType));
+                        break;
+                    default:
+                        res.set('content-type', contentType);
+                        res.end(iconv.encode(src, charsetDefault));
+                }
+            });
+
+        destServer = app.listen(2000);
+
+        session                    = new Session();
+        session.getAuthCredentials = function () {
+            return null;
+        };
+        session.id                 = 'sessionId';
+
+        proxy = new Proxy(proxyHostname, proxyPort1, proxyPort2);
+        proxy.openSession('http://127.0.0.1:2000/', session);
+    });
+
+    after(function () {
+        destServer.close();
+        proxy.close();
+        requestAgent.resetKeepAliveConnections();
+    });
+
+    // Tests
+    describe('Pages', function () {
+        var charsetTestMatch  = new RegExp('<meta class="' + SHARED_CONST.SHADOW_UI_CHARSET_CLASSNAME + '" charset="([^\"]+)"', '');
+        var charsetTestFunc   = function (originUrl, expectedCharset, expectedBody, callback) {
+            var url     = urlUtil.getProxyUrl('http://127.0.0.1:2000' +
+                                              originUrl, proxyHostname, proxyPort1, sessionId);
+            var options = {
+                url:     url,
+                headers: {
+                    accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,*!/!*;q=0.8'
+                }
+            };
+
+            request(options, function (err, res, body) {
+                var decodedBody = iconv.decode(new Buffer(body), expectedCharset);
+                var match       = decodedBody.match(charsetTestMatch);
+                var charset     = match && match[1];
+
+                expect(charset).eql(expectedCharset);
+                compareCode(body, expectedBody);
+
+                callback();
+            });
+        };
+        var emptyFunc         = function () {
+        };
+        var getExpectedSrcStr = function (src, charset, addBOM) {
```

Comment:
`getExpectedStr`. BTW I find it quite hard to parse this method. Please, refactor it.